### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by techmoves**
in this repo i did this:

- [x] created initial state

- [x] created add book and remove-book function .
- [x] Display your books, received from the slice, in a reusable component

- [x] Dispatch actions using useDispatch

- [x] Add a <Button> component, which includes:

    - An event handler that adds a book to the books array (with attributes id, title and author)

- [x] Add a <Button> component, which includes:

  - An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/BooksItem.jsx [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import React from 'react';
 import PropTypes from 'prop-types';
```
Comment (Gambit142) [inline src/components/BooksItem.jsx:2]: ### Although I am highlighting this piece of code, the change I am about to request should be done in the `src/components/BooksList.js` file. 

- [x] Kindly use the `useSelector` hook to access the store and retrieve the books reducer in that file. This means that you would have to remove the **_static array_** you created in that file and replace it with the `initialState` array gotten from the books reducer in the redux slice file for the books. Kindly implement this change. 👍 💻 

#### Hint: 
```JavaScript
import { useSelector } from 'react-redux';
const books = useSelector((state) => state.book.bookItem);
```

File: src/redux/books.js/booksSlice.js [hunk 2]
```diff
@@ -30,11 +30,7 @@ const booksSlice = createSlice({
       state.bookItem.push(action.payload);
     },
     removeBook: (state, action) => {
-      const bookId = action.payload;
-      return {
-        ...state,
-        bookIt25056em: state.bookItem.filter((book) => book.id !== bookId),
-      };
+      state.books = state.books.filter((book) => book.id !== action.id);
     },
```
Comment (Gambit142) [inline src/redux/books.js/booksSlice.js]: - [x] When I try to delete a book, I get this error:

image

There is no `state.books` variable/object. Remember that initialState contains an object named `bookItem` which contains an array of books. Therefore, to access it, you need to make your statement like this:

```JavaScript
state.bookItem = state.bookItem.filter((book) => book.item_id !== action.payload);
```

Please ensure that the application can delete a particular book one at a time. 👍 💻

File: src/components/BooksItem.jsx [hunk 3]
```diff
@@ -1,7 +1,11 @@
...
+import { useDispatch } from 'react-redux';
+
+import { removeBook } from '../redux/books.js/booksSlice';
 
 export default function BooksItem({ list }) {
```
Comment (Gambit142) [inline src/components/BooksItem.jsx:7]: - [OPTIONAL] The way you defined the props type for the BooksItem component is incorrect and it produces this error in the console:

image

You can read this [article](https://legacy.reactjs.org/docs/typechecking-with-proptypes.html) or [this article](https://www.freecodecamp.org/news/how-to-use-proptypes-in-react/) to understand the different prop types and how to shape up the correct prop types for each variable. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @techmoves 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Use of redux actions in React components ✔️ 
- Use of Provider component to wrap main React component ✔️

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

### **_You can also consider the comment tagged [OPTIONAL] under this review summary. 👍 💻_** 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/books.js/booksSlice.js (hunk 2 above)
Comment (techmoves) [reply src/redux/books.js/booksSlice.js]: thank you so much resoved

File: src/components/BooksItem.jsx (hunk 1 above)
Comment (techmoves) [reply src/components/BooksItem.jsx:2]: thank you so much resoved

**Review by rustam-tolipov — APPROVED** [review]
Hi @techmoves, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. App is working as expected 👍 
4. Great readme 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**